pkg/registries/docker/metadata_v2.go

```diff
@@ -102,8 +103,20 @@ func HandleOCIManifest(r *Registry, remote, ref string) (*storage.ImageMetadata,
 	return &storage.ImageMetadata{
 		V1: v1Metadata,
 		V2: &storage.V2Metadata{
-			Digest: ref,
+			Digest: imageReference(ref, dig),
 		},
 		LayerShas: layers,
 	}, nil
 }
+
+// imageReference will return digest if it is populated and ref is not a digest.
+// Otherwise returns ref.
+func imageReference(ref string, digest godigest.Digest) string {
+	if digest != "" {
+		if _, err := godigest.Parse(ref); err != nil {
+			return string(digest)
+		}
+	}
+
+	return ref
+}
```

Comment:
Maybe clearer:

```suggestion
// digestOrRef returns an image digest if found in the reference, then if found in the digest parameter.
// If no digest is found, return the reference as is.
func digestOrRef(ref string, digest godigest.Digest) string {
	_, err := godigest.Parse(ref)
	if err != nil && digest != "" {
		return string(digest)
	}	
	return ref
}
```

Reply:
I'll update the method name / comment, performing `godigest.Parse()` when digest == "" is wasteful however - rather leave that to be conditional.

Reply:
https://github.com/stackrox/stackrox/pull/11084/commits/f9ac114e1d571999cb0bea98e9d503ee2b4b47c7

Reply:
Changed up the logic to be clearer.